openmc/lib/cell.py

```diff
@@ -0,0 +1,228 @@
+_dll.openmc_cell_set_id.errcheck = _error_handler
+_dll.openmc_cell_set_temperature.argtypes = [
+    c_int32, c_double, POINTER(c_int32)]
+_dll.openmc_cell_set_temperature.restype = c_int
+_dll.openmc_cell_set_temperature.errcheck = _error_handler
+_dll.openmc_get_cell_index.argtypes = [c_int32, POINTER(c_int32)]
+_dll.openmc_get_cell_index.restype = c_int
+_dll.openmc_get_cell_index.errcheck = _error_handler
+_dll.cells_size.restype = c_int
+_dll.openmc_cell_bounding_box.argtypes = [c_int,
+                                          POINTER(c_double),
+                                          POINTER(c_double)]
+_dll.openmc_cell_bounding_box.restype = c_int
+_dll.openmc_cell_bounding_box.errcheck = _error_handler
+
+
+class Cell(_FortranObjectWithID):
+    """Cell stored internally.
+
+    This class exposes a cell that is stored internally in the OpenMC
+    library. To obtain a view of a cell with a given ID, use the
+    :data:`openmc.lib.cells` mapping.
+
+    Parameters
+    ----------
+    index : int
+         Index in the `cells` array.
+
+    Attributes
+    ----------
+    id : int
+        ID of the cell
+
+    """
+    __instances = WeakValueDictionary()
+
+    def __new__(cls, uid=None, new=True, index=None):
+        mapping = cells
+        if index is None:
+            if new:
+                # Determine ID to assign
+                if uid is None:
+                    uid = max(mapping, default=0) + 1
+                else:
+                    if uid in mapping:
+                        raise AllocationError('A cell with ID={} has already '
+                                              'been allocated.'.format(uid))
+
+                index = c_int32()
+                _dll.openmc_extend_cells(1, index, None)
+                index = index.value
+            else:
+                index = mapping[uid]._index
+
+        if index not in cls.__instances:
+            instance = super().__new__(cls)
+            instance._index = index
+            if uid is not None:
+                instance.id = uid
+            cls.__instances[index] = instance
+
+        return cls.__instances[index]
+
+    @property
+    def id(self):
+        cell_id = c_int32()
+        _dll.openmc_cell_get_id(self._index, cell_id)
+        return cell_id.value
+
+    @id.setter
+    def id(self, cell_id):
+        _dll.openmc_cell_set_id(self._index, cell_id)
+
+    @property
+    def name(self):
+        name = c_char_p()
+        _dll.openmc_cell_get_name(self._index, name)
+        return name.value.decode()
+
+    @name.setter
+    def name(self, name):
+        name_ptr = c_char_p(name.encode())
+        _dll.openmc_cell_set_name(self._index, name_ptr)
+
+    @property
+    def fill(self):
+        fill_type = c_int()
+        indices = POINTER(c_int32)()
+        n = c_int32()
+        _dll.openmc_cell_get_fill(self._index, fill_type, indices, n)
+
+        if fill_type.value == 1:
+            if n.value > 1:
```

Comment:
To confirm, `n.value` should never equal 0?

Reply:
Correct, it should always be positive